subt_ign/src/GameLogicPlugin.cc

```diff
@@ -640,6 +644,31 @@ void GameLogicPlugin::PostUpdate(
     std::map<std::string, ignition::math::Pose3d>::iterator baseIter =
       this->dataPtr->poses.find(subt::kBaseStationName);
 
+    _ecm.Each<gazebo::components::DetachableJoint>(
+        [&](const gazebo::Entity &,
+            const gazebo::components::DetachableJoint *_detach) -> bool
+        {
+          auto parentModel = _ecm.Component<gazebo::components::ParentEntity>(
+              _detach->Data().parentLink);
+          auto childModel = _ecm.Component<gazebo::components::ParentEntity>(
+              _detach->Data().childLink);
+
+          auto parentName = _ecm.Component<gazebo::components::Name>(
+              parentModel->Data());
+
+          auto childName = _ecm.Component<gazebo::components::Name>(
+              childModel->Data());
+          // Store the marsupial pairs, and handle the special case of
+          // "X1_platform"
+          if (parentName->Data() != "X1_platform" &&
+              childName->Data() != "X1_platform")
```

Comment:
The platform is actually going to be named `<name>_platform`, so maybe check if the name ends with `_platform`? If we filter out non-performers, we won't have to do this check.

Reply:
I think this fixes it: 9d1b038f198dab3f825d7780e6503e657076ab51